client/snap_op.go

```diff
@@ -34,6 +35,19 @@ func (client *Client) InstallSnap(name, channel string) (string, error) {
 	return client.doAsync("POST", path, nil, body)
 }
 
+// SideloadSnap sideloads the snap with the given path, returning the UUID
+// of the background operation upon success.
+//
+// XXX: add support for "X-Allow-Unsigned"
+func (client *Client) SideloadSnap(path string) (string, error) {
+	f, err := os.Open(path)
```

Comment:
Actually, sorry. This os.File will be handed down to http.NewRequest which will close the file automatically.